README.md

```diff
@@ -1 +1,163 @@
-# url_shortner
+# Shortit - URL Shortener
+
+[![License](https://img.shields.io/badge/license-MIT-blue.svg)](LICENSE)
+
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Features](#features)
+- [Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Installation](#installation)
+  - [Running the Project](#running-the-project)
+- [Usage](#usage)
+  - [API Endpoints](#api-endpoints)
+  - [Example Request/Response](#example-requestresponse)
+- [Project Structure](#project-structure)
+- [Technologies](#technologies)
+- [Contributing](#contributing)
+- [License](#license)
+- [Contact](#contact)
+
+## Introduction
+
+**Shortit** is a URL shortening service that allows users to generate custom and compact URLs from long, cumbersome links. Users can create unique short URLs with custom aliases, making it ideal for sharing across platforms.
+
+The project is built using **Node.js**, **Express.js**, **MongoDB**, and includes custom error handling and validation with `express-validator` and custom middleware. The service allows both standard and customized URL shortening, with a simple and scalable backend structure.
+
+## Features
+
+- Shorten any URL.
+- Custom alias support (e.g., `shortit/custom-alias`).
+- Automatically generate random short URLs.
+- Track original URLs and their shortened counterparts.
+- Detailed error handling for invalid or duplicate entries.
+- Easy-to-use API for integration with other services.
+
+## Getting Started
+
+### Prerequisites
+
+Ensure you have the following installed on your local machine:
+
+- Node.js (version 14 or later)
+- MongoDB (version 4 or later)
+
+### Installation
+
+1. Clone the repository:
+   \`\`\`bash
+   git clone https://github.com/mosidrum/url_shortner.git
+   cd url_shortner
+   \`\`\`
+
+2. Install the required dependencies:
+   \`\`\`bash
+   npm install
+   \`\`\`
```

Comment:
This should be something like this. It does not display well in the markdown.

```bash
npm install
```

Btw, you could simply do `npm install`. Apply this everywhere necessary